experimental/crabbake/src/lib.rs

```diff
@@ -0,0 +1,156 @@
+extern crate proc_macro;
+mod bakeable;
+use bakeable::Bakeable;
+use icu_provider::iter::IterableDataProviderCore;
+use icu_provider::prelude::*;
+use icu_provider_fs::FsDataProvider;
+use proc_macro::TokenStream;
+use proc_macro2::TokenStream as TokenStream2;
+use quote::{format_ident, quote};
+use syn::Ident;
+
+#[proc_macro]
+pub fn static_data_provider(_input: TokenStream) -> TokenStream {
```

Comment:
crabbake should not know anything about data providers, this should happen at a separate layer. You should be defining a custom derive  instead

also the trait cannot live in the same crate as the custom derive, use `crabbake/derive`

Reply:
(we may want the static data provider generation to be something the datagen tool does, too, but having a proc macro for it is decent too)

Reply:
I know, I was just playing around with this and didn't want to throw this code away.

How would the datagen work? It would create a .rs file? That seems more brittle than a proc macro.

Reply:
Ah, I see.

I think folks might want both, potentially. They can use the same code.

But for now just having const data provider seems ok